docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,624 @@
+qdrant_qa = load_dataset("atitaarora/qdrant_doc_qna", split="train")
+
+questions = []
+answers = []
+contexts = []
+ground_truths = []
+
+for query in qdrant_qa:
+    question = query["question"]
+    questions.append(question)
+    ground_truths.append([query["answer"]])
+    
+    rag_response = query_with_context(question)
+    answers.append(rag_response)
+    search_result_context = client.query(collection_name=COLLECTION_NAME, query_text=question,limit=4)
+    context = ["document:"+r.document+",source:"+r.metadata['source'] for r in search_result_context]
+    contexts.append(context)    
+
+rag_response_data = {
+    "question": questions,
+    "answer": answers,
+    "contexts": contexts,
+    "ground_truths": ground_truths
+}
+
+## This dataset 'rag_response_dataset', is created to evaluate the RAG system using the RAGAS format.
+## Note: this dataset is specifically for evaluation purposes. Therefore, it needs to be recreated every time changes are made to the RAG config.
+rag_response_dataset = Dataset.from_dict(rag_response_data)
+# The dataset is then exported as a CSV file, with a filename that includes details of the experiment for easy identification, such as the chunk size used in this case -  rag_response_dataset.to_csv('rag_chunk_512.csv')
+```
+
+The subroutine above uses an abstraction of the RAG pipeline method `query_with_context()`. The `rag_response_dataset` uses an older format in which `ground_truths` is a list of strings, while the latest format uses `ground_truth` as a single string value (representing the correct answer to a question). 
+We will show a sample naive-rag example in a bit.
+You can choose your preferred format. 
+
+Now that we have a baseline as well as the subroutine to create evaluation dataset .
+
+Let us understand what all can you measure with RAGAS.
+
+## Metrics
+
+It is a good idea to check out [core concepts](https://docs.ragas.io/en/latest/concepts/index.html) of RAGAS , at this stage.
+
+RAGAS is focused on retrieval and generation stages of RAG along with End-to-End evaluation of RAG and it relies on concept of [metrics driven development](https://docs.ragas.io/en/latest/concepts/metrics_driven.html)
+
+![../assets/use_cases/retrieval_augmented_generation_eval_qdrant_ragas/ragas_metrics.png](../assets/use_cases/retrieval_augmented_generation_eval_qdrant_ragas/ragas_metrics.png)
+
+### RAGAS's key evaluation metrics
+
+The key evaluation metrics offered by RAGAS are:
+
+- [Faithfulness](https://docs.ragas.io/en/latest/concepts/metrics/faithfulness.html) - Scaled between 0 and 1, it measures factual consistency of the answer. The preference given to faithfulness score being closer to 1 signifying higher consistency. It is measured as :
+
+```notion
+Faithfulness score = (Number of claims in the generated answer that can be inferred from given context) / 
+										 (Total number of claims in the generated answer)
+```
+
+and uses context and answer fields from .
```

Comment:
@atarora 
can you clarify what  "and uses context and answer fields from"?
this means Faithfulness uses context and answer fields from Answer relevancy...? Or you mean RAGAS other key metrics include context and relevancy fields from Answer relevancy and Context recall...?

Reply:
looks like a misplaced `from` to me , thanks for pointing out !